CMakeLists.txt

```diff
@@ -39,13 +46,8 @@ target_link_libraries(${PROJ_NAME} PUBLIC
     cli
     littlefs
     cmake_git_version_tracking
-    hardware_i2c
-    hardware_spi
-    hardware_flash
-    hardware_adc
     cmsis_core
```

Comment:
We can move `cmsis_core` to the `board_hal` library as well, since other MCUs/boards might not be ARM

Reply:
It's not actually being used right now. I'm just going to remove it.

Reply:
For real? I wonder why I had it in there. I remember adding it for some reason, can't think of why. Okay we'll nuke it for now